app/clients/lifx_client.rb

```diff
@@ -0,0 +1,79 @@
+class LifxClient
+  include HTTParty
+  base_uri 'https://api.lifx.com/v1/lights'
+  
+  def initialize(auth_token, selector)
+    @auth_token = auth_token
+    @selector = selector
+  end
+    
+  def get_selectors
+    response = self.class.get("/all", 
+      headers: authorization_header,
+    )
+    if response.code == 404
+      return false
+    else
+      lights_json = JSON.parse(response.body)
+      lights = []
+      groups = []
+      lights_json.each do |light| 
+        lights << Light.new(light["id"], light["label"])
```

Comment:
I like your suggestion. It's more looping, but way easier to read and there hopefully won't be thousands of lights :wink: